### ISSUE TITLE
can we generate rst-support.rst, rather than hand maintaining it?

### ISSUE THREAD

**Issue by elliefm** [cyrusimap/cyrus-imapd#4351]
`docsrc/imap/rfc-support.rst` contains the list of RFCs we support, and becomes http://www.cyrusimap.org/imap/rfc-support.html when the website is built.  We're not perfect at keeping it up to date...

Perhaps we could generate this source file instead.  We already have a fairly structured list of most? all? this stuff under `static struct capa_struct base_capabilities[]` in `imap/imapd.c`, and this structure will get better when #4349 lands.  And we already do this sort of thing elsewhere, for example `docsrc/imap/reference/manpages/configs/imapd.conf.rst` is generated from `lib/imapoptions`.

I haven't looked really closely to see if there's perfect overlap between the one and the other, but if the overlap is near enough, and as long as we're strict about how we maintain the `base_capabilities[]` array, I don't think it'd be hard to make a parser/generator to reduce the double-handling.

**Comment by elliefm** [comment]
(Assigning this to myself cause it seems fun and I want to not forget it, but don't feel bad stealing it from me if you get to it first.)

**Comment by dilyanpalauzov** [comment]
I do not think the file can be generated.  For Sieve and JMAP the list of supported RFCs is not as good organized in the source code, as for IMAP.  This can be of course improved.

The main advantage of the separate rst-support.rst file currently is, that it can contain notes about how a particular RFC is implemented, like having only some requerements implemented and others not.  Or if the RFC leaves choices to the implementation, writing down what choices were taken.

I would prefer if in the future there are more notes included in rfc-support, e.g. what is implemented from JMAP and what is missing.

**Comment by rjbs** [comment]
I agree with Dilyan.  I don't think this can be automated well.  We'll have to get better at keeping it up to date.  (Maybe a comment in the changes template about "does it add new RFC support?)
